number/src/baby_bear.rs

```diff
@@ -0,0 +1,467 @@
+use p3_baby_bear::BabyBear;
+use schemars::{
+    schema::{Schema, SchemaObject},
+    JsonSchema,
+};
+use serde::{Deserialize, Serialize};
+
+use num_traits::{ConstOne, ConstZero};
+use std::ops::{Add, AddAssign, Div, Mul, MulAssign, Neg, Not, Sub, SubAssign};
+use std::str::FromStr;
+use std::{collections::BTreeSet, fmt::LowerHex};
+
+use crate::{BigUint, FieldElement, KnownField, LargeInt};
+use ark_ff::{One, Zero};
+
+use core::fmt::{self, Debug, Formatter};
+use core::hash::Hash;
+
+use p3_field::{AbstractField, Field, PrimeField32};
+
+#[derive(
+    Debug,
+    Copy,
+    Clone,
+    Default,
+    Eq,
+    Hash,
+    PartialEq,
+    Ord,
+    PartialOrd,
+    Serialize,
+    Deserialize,
+    derive_more::Display,
+)]
+pub struct BabyBearField(BabyBear);
+
+const P: u32 = 0x78000001;
+const MONTY_BITS: u32 = 32;
+const MONTY_ZERO: u32 = to_monty(0);
+const MONTY_ONE: u32 = to_monty(1);
+
+const fn to_monty(x: u32) -> u32 {
+    (((x as u64) << MONTY_BITS) % P as u64) as u32
```

Comment:
Didn't we ditch montgomery? ;)

Reply:
This is all copied from Plonky3, our type is just a wrapper over their type. I had to copy this directly here because `const BabyBear::new()` is private within the `p3_field` crate, as well as this function, and I needed it for the `ConstZero` and `ConstOne` traits we need to implement here.